apps/omg_child_chain_rpc/test/omg_child_chain_rpc/web/plugs/health_test.exs

```diff
@@ -93,7 +101,10 @@ defmodule OMG.ChildChainRPC.Plugs.HealthTest do
   defp pull_client_alarm(0, _, _), do: :cant_match
 
   defp pull_client_alarm(n, match, fnn) do
-    case fnn.() do
+    endpoint_call_result = fnn.()
+    match = Map.put_new(match, "version", Map.get(endpoint_call_result, "version"))
```

Comment:
Oh, we just dealing with version to be able to use: `case ... do ^match ->` and again paying to much attention to the response details.
I saw @pthomalla proposition, after all concerns with naming overall it sees to much cleaner and more importantly less prone to internal details of the response, e.g.: https://github.com/omisego/elixir-omg/pull/860/files#r304838025

Reply:
@pnowosie again, this is a unit test. it tests a very specific response, because that's what it was written for. I want to test if I'm getting that *EXACT* response.

Reply:
also, take a deeper look, this is a _plug_ test. that's why this _plug_ was written for. for this exact purpose.

Reply:
>  I want to test if I'm getting that EXACT response
Why? Isn't this plug handle also failure of `/submit` endpoint? Why then you've chosen `/block.get`.
IMHO we're testing here that `because alarm was raised no request is possible to fulfill`
being too specific we coupling exact error messages with `Health` plug

Reply:
Because this plug kicks in before any controllers are being called. And because of that I don't need to test on all routes. 
`Health` plug produces that error message.
By definition, a *unit test*, is a coupling of the *test* and the *unit*.

Reply:
Ok, I missed that this's plug response, probably test title should not expose `/block.get` but say `all routes`.
Then I agree that **it's OK to specifically check error's code & description** (which is set by the plug).

Why then we require exact there will be `version` and all other fields in the response. Error response format is not `unit` we're testing here. Do you agree?

Reply:
I simply followed how controller's are testing their response - https://github.com/omisego/elixir-omg/blob/master/apps/omg_child_chain_rpc/test/omg_child_chain_rpc/web/controllers/block_test.exs#L26-L38

Reply:
and there we have matching no equality written in a strange way

Reply:
Do it's just me or are we talking the same thing? The [example you mentioned](https://github.com/omisego/elixir-omg/blob/master/apps/omg_child_chain_rpc/test/omg_child_chain_rpc/web/controllers/block_test.exs#L26-L38).
Where are there `version` or `object => error` in the controller tests?

Reply:
The version in this case doesn't really matter, because whatever is returned from the server is considered valid.
`object => error` is here https://github.com/omisego/elixir-omg/blob/master/apps/omg_child_chain_rpc/test/omg_child_chain_rpc/web/controllers/block_test.exs#L29

Reply:
Don't you think then that function @pthomalla introduced was more intuitive. As he observed here we're checking **equality** not a match.

Also I meant we shouldn't be such specific in the controller tests.

Reply:
No. I do not think his approach is more intuitive. I don't think it's making things better. It's actually making things worse by coupling :omg with :omg_child_chain_rpc and removing the test being self contained and it's not addressing any issues.

Reply:
„Yet the wise men persisted in their resolution.” This is fine, I’m not going to burn more gas 🤓

✅

Reply:
Address that this test was broken, that match is equality and that this name suggests that getting alarm is call ones.
As to the location of the function, We can move it to a more appropriate place.
#860

Reply:
I don’t think I am able to add anything to this discussions. I’ve said everything I had.